### PR TITLE
feat: 569 apply facets as filters to what is being currently searched

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
Upon clicking facets for papers do not call XDD to requery, Instead just filter what results are already grabbed and update new facets on left page

https://user-images.githubusercontent.com/17088680/215792231-42630d48-4b2a-4c0a-95f3-ea666405d5f7.mov

Resolves #(issue)
#569

### CODE REVIEW

**Comment by YohannParis** [comment]
Looks great, but why do you have a loading spinner if the objects are filtered locally?

File: packages/client/hmi-client/src/services/data.ts [hunk 1]
```diff
@@ -158,96 +192,137 @@ const getAssets = async (
...
-							(conceptResult) =>
-								conceptResult.name === conceptNameOrCurie ||
-								conceptResult.curie === conceptNameOrCurie
-						);
-						// update the clause value by mapping concept/curie to model id
-						clauseValues.push(...matching.map((m) => m.id));
-						curies.push(...matching.map((m) => m.curie));
+	if (searchParam) {
+		// Filtering for model/dataset data
+		if (searchParam && searchParam.filters && searchParam.filters.clauses.length > 0) {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/data.ts:197]: You can use lodash [`isEmpty()`](https://lodash.com/docs/4.17.15#isEmpty)
```js
import { isEmpty } from 'lodash';
...
if (!isEmtpy(searchParam?.filters?.clauses) {
    ...
```

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/data.ts:197]: Awesome! Ill fix this
Thanks for the tip I will have to keep this library in mind

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/data.ts]: Unfortunately I still need the first two arguments for the code inside to not freak out thinking the object could be empty.

**Comment by Tom-Szendrey** [comment]
> Looks great, but why do you have a loading spinner if the objects are filtered locally?

I had not touched the loader spinner. It exists for the models and dataset filtering as well. The other two are smaller objects and smaller sets. They may also be filtered more efficiently im not certain
Is this something you want me to remove?

File: packages/client/hmi-client/src/services/data.ts [hunk 2]
```diff
@@ -158,96 +192,137 @@ const getAssets = async (
...
-			const finalAssetIDs = assetResults.map((m) => m.id);
-			conceptFacets.results.forEach((conceptFacetResult) => {
-				if (finalAssetIDs.includes(conceptFacetResult.id)) {
-					curies.push(conceptFacetResult.curie);
-				}
-			});
-		}
+			// re-create the concept facets if the user has applyied any concept filters
+			const uniqueCuries = uniq(curies);
+			if (uniqueCuries.length > 0) {
```
Comment (shawnyama) [inline packages/client/hmi-client/src/services/data.ts:261]: I guess you could use the `isEmpty` in these other checks too

**Review by shawnyama — COMMENTED** [review]
Lookin good

**Review by shawnyama — APPROVED** [review]
I'll see about dealing with the spinner on my end

**Pull request merged**